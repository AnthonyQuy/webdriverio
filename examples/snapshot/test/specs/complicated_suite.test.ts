import { expect, browser, $ } from '@wdio/globals'

describe('Level 1 suite', () => {
    it('test1 1', async () => {
        await browser.url('https://the-internet.herokuapp.com/login')

        await $('#username').setValue('tomsmith')
        await $('#password').setValue('SuperSecretPassword!')
        await $('button[type="submit"]').click()

        await expect($('#flash')).toBeDisplayed()

        await expect({ a: 'a' }).toMatchSnapshot()
    })
    describe('Level 2 suite', () => {
        it('test2 2', async () => {
            await expect({ a: 'a' }).toMatchSnapshot()
        })
        describe('Level 3 suite', () => {
            it('test 3', async () => {
                await expect({ a: 'a' }).toMatchSnapshot()
                const element = await $('#flash')
                const html = await element.getHTML()
                await expect(html).toMatchSnapshot()
                await expect($('#flash')).toMatchElementSnapshot()
            })
        })
    })
})
