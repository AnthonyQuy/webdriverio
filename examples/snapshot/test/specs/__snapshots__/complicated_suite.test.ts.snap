// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 1 suite > Level 2 suite > Level 3 suite > test 3 1`] = `
Object {
  "a": "a",
}
`;

exports[`Level 1 suite > Level 2 suite > Level 3 suite > test 3 2`] = `
"<div data-alert=\\"\\" id=\\"flash\\" class=\\"flash success\\">
            You logged into a secure area!
            <a href=\\"#\\" class=\\"close\\">×</a>
          </div>"
`;

exports[`Level 1 suite > Level 2 suite > Level 3 suite > test 3 3`] = `
"<div data-alert=\\"\\" id=\\"flash\\" class=\\"flash success\\">
            You logged into a secure area!
            <a href=\\"#\\" class=\\"close\\">×</a>
          </div>"
`;

exports[`Level 1 suite > Level 2 suite > test2 2 1`] = `
Object {
  "a": "a",
}
`;

exports[`Level 1 suite > test1 1 1`] = `
Object {
  "a": "a",
}
`;
